src/PluginManager.js

```diff
@@ -0,0 +1,29 @@
+// @flow
+/**
+ * This file contains the core logic for handling KaTeX plugins.
+ * It provides utility functions to add plugins to the library.
+ * PluginManager exposes main KaTeX functions as a compiler object to each plugin.
+ * These functions are usually closures that change a special object
+ * in their corresponding modules. These special objects are used
+ * to make decisions, find symbols and font metrics, call TeX functions, etc.
+ * So basically by calling these compiler functions,
+ * plugins can extend or change library's behaviour at the runtime.
+ */
+import { defineSymbol } from './symbols';
+import { defineFunction } from './functions';
+import { addFontMetrics } from './fontMetrics';
+
+const compiler = {
+    defineSymbol,
+    defineFunction,
+    addFontMetrics,
```

Comment:
Why not just expose these functions directly on the `katex` object?  There isn't really a concept of plugins internally.  I think `defineSymbol` is relatively stable, but `defineFunction` requires access to a lot of internal classes to be useful.

Reply:
These functions are basically internal utilities of KaTeX.  
I did not think exposing them to normal users was a good idea on the other hand advanced users can use them without defining a new javascript class.

Reply:
We can expose them with a `__` underscore prefix to indicate that they should be used with caution.

Reply:
Yes I have done this.   
Please checkout my next comment.